kernel_gateway/gatewayapp.py

```diff
@@ -30,16 +32,32 @@ class KernelGatewayApp(JupyterApp):
         Provisions kernels and bridges Websocket communication
         to/from them.
     '''
+    port_env = 'KG_PORT'
+    port = Integer(config=True,
+        help="Port on which to listen (KG_PORT env var)"
+    )
+    def _port_default(self):
+        return int(os.getenv(self.port_env, 8888))
 
-    port = Integer(8888, config=True,
-        help="The port the notebook server will listen on."
+    ip_env = 'KG_IP'
+    ip = Unicode(config=True,
+        help="IP address on which to listen (KG_IP env var)"
     )
+    def _ip_default(self):
+        return os.getenv(self.ip_env, '0.0.0.0')
```

Comment:
I'm never a fan of services that listen on all interfaces by default, but that's also the easiest to work with in the docker case.